web/web.go

```diff
@@ -224,9 +225,11 @@ func (h *Handler) Quit() <-chan struct{} {
 	return h.quitCh
 }
 
-// Reload returns the receive-only channel that signals configuration reload requests.
-func (h *Handler) Reload() <-chan struct{} {
-	return h.reloadCh
+// Reload returns two cahnnels:
+//   - receive-only channel that signals configuration reload requests.
+//   - send-only channel with errors happened during configuration reload.
```

Comment:
The fundamental problem is, that the error or success you receive is not tied to your request.
Two requests can trigger a configuration reload and get the response of the other one respectively.

There are different solutions to the problem. Busy right now but I'll try to think about a sane approach.